src/Network/Minio/PutObject.hs

```diff
@@ -66,47 +67,61 @@ putObjectInternal b o (ODFile fp sizeMay) = do
 
   case finalSizeMay of
     -- unable to get size, so assume non-seekable file and max-object size
-    Nothing -> sequentialMultipartUpload b o (Just maxObjectSize) $
+    Nothing -> sequentialMultipartUpload b o opts (Just maxObjectSize) $
                CB.sourceFile fp
 
     -- got file size, so check for single/multipart upload
     Just size ->
       if | size <= 64 * oneMiB -> either throwM return =<<
-           withNewHandle fp (\h -> putObjectSingle b o [] h 0 size)
+           withNewHandle fp (\h -> putObjectSingle b o (pooToHeaders opts) h 0 size)
          | size > maxObjectSize -> throwM $ MErrVPutSizeExceeded size
-         | isSeekable -> parallelMultipartUpload b o fp size
-         | otherwise -> sequentialMultipartUpload b o (Just size) $
+         | isSeekable -> parallelMultipartUpload b o opts fp size
+         | otherwise -> sequentialMultipartUpload b o opts (Just size) $
                         CB.sourceFile fp
 
-parallelMultipartUpload :: Bucket -> Object -> FilePath -> Int64
-                        -> Minio ETag
-parallelMultipartUpload b o filePath size = do
+parallelMultipartUpload :: Bucket -> Object -> PutObjectOptions
+                        -> FilePath -> Int64 -> Minio ETag
+parallelMultipartUpload b o opts filePath size = do
   -- get a new upload id.
-  uploadId <- newMultipartUpload b o []
+  uploadId <- newMultipartUpload b o (pooToHeaders opts)
 
   let partSizeInfo = selectPartSizes size
 
-  -- perform upload with 10 threads
-  uploadedPartsE <- limitedMapConcurrently 10
-                    (uploadPart uploadId) partSizeInfo
+  case pooNumThreads opts of
```

Comment:
this can be factored as,
```hs
let threads = fromMaybe 10 $ pooNumThreads opts
```

Reply:
Yep thanks, another deliberate attempt to get answers :P